### DISCUSSION TITLE
Comparing Resources using FluentAssertions

### DISCUSSION THREAD

**Discussion by LodewijkSioen** [FirelyTeam/firely-net-sdk#2498] [Q&A] [answered]
So I'm writing integration tests. I want to compare what's returning from the server to what I'm expecting using [Fluent Assertions](https://fluentassertions.com). I want to exclude the `Id` and other properties that are non deterministic.

However, since `Base` is kind of an `IDictionary`, FluentAssertions keeps wanting to compare the keys and values in the dictionary.

```
(new Patient {Id = "123", Active = true} as object)
        .Should().BeEquivalentTo(new Patient {Active = true}, 
            o => o.Excluding(p => p.Id));
```
Results in:
```
Expected (new Patient {Id = "123", Active = true} as object) to be a collection with 1 item(s), but {[id, 123], [active, true]}"
"contains 1 item(s) more than"
"{[active, true]}.
```

Any known workarounds for this?

**Comment by ewoutkramer** [comment, marked as answer]
Yeah, it took me ages before I understood the fine details of FluentAssertions in this regard. It actually works much better if you use `BeEquivalentTo()` with *anonymous* objects, so e.g.

```
Patient p = new()
{
	Deceased = new FhirBoolean(true)
};

p.Should().BeEquivalentTo(new { Deceased = new FhirBoolean(true) });
```

Note the `new {}`, not `new Patient {}`.

**Reply by LodewijkSioen** [reply]
Yeah, it's the fact that each resource is also a Collection that confuses the equivalency algoritm. Anonymous objects bypasses that problem.
However, I've since switched to [Verify](https://github.com/VerifyTests/Verify/) to do snapshot testing, which is a much simpeler pattern.

**Reply by ewoutkramer** [reply]
We use that approach to test the intermediate steps in our new validator, though I did write something of my own. I should be using this library though.

Again, this looks like one more proof that letting the poco's implement IDictionary/IEnumerable<Kvp> directly was a mistake. I dont like the debug experience in vs either. Maybe we should break this in 6.0 and introduce AsIDictionary on pocos.
